### DISCUSSION TITLE
Hook은 언제 필요한 걸까요?

### DISCUSSION THREAD

**Discussion by woowapark** [woowacourse/react-lotto#39] [토론해보기]
- 리액트에서 Hook은 무엇인가요?
- Hook은 언제 필요한 걸까요?
- 리액트 팀에서 말하는 필요성에 대해 공감이 되나요? 어떻게 생각하나요?

**Comment by jho2301** [comment]
함수컴포넌트에서 상태가 필요할 때!

**Reply by zereight** [reply]
Hook도 단점이 있나요

**Reply by 365kim** [reply]
음 최상위에 써야한다. 반복문, 조건문 내 쓰면 안된다는 등의 조건을 지키기 위해 신경써야 한다는 점도 단점이 될 수 있으려나용

**Comment by seojihwan** [comment]
함수컴포넌트에서 라이프 사이클의 일부, 메모이제이션을 사용하기 위해 ?

**Comment by Puterism** [comment]
상위 컴포넌트에서 특정 컴포넌트에서만 필요한 상태까지 과도하게 가지고 있는 경우, 이를 분리하기 위해서

**Comment by zigsong** [comment]
클래스 컴포넌트들에 있던 상태와 라이프사이클 메소드를 사용하기 위해서

**Comment by goni-ssi** [comment]
life cycle 관련 메서드를 클래스 컴포넌트에서는 모두 한 곳에서 관리하지만 함수형 컴포넌트에서는 분리하여 관리할 수 있어서 좋았습니다 :)

**Comment by bucketHaneul** [comment]
함수형 컴포넌트에서 부모의 props로 받는 것 말고 나만의 상태를 가지고 싶을 때
